kratos/solving_strategies/schemes/residual_based_adjoint_static_scheme.h

```diff
@@ -237,6 +309,7 @@ class ResidualBasedAdjointStaticScheme : public Scheme<TSparseSpace, TDenseSpace
 
     AdjointResponseFunction::Pointer mpResponseFunction;
     std::vector<LocalSystemVectorType> mAdjointValues;
+    std::vector<LocalSystemMatrixType> mLHS;
```

Comment:
I don't get it. Why do you need to store the LHS? The LHS is always computed in the function it's used in, and the stored value is never read outside those functions.

In my opinion, it would make more sense to declare local variables for the left hand sides, and dispose of them after the contribution is computed. Not only would it make the code cleaner, but it would also get rid of all `OpenMPUtils::ThisThread` calls.